src/fiatconnect/saga.ts

```diff
@@ -206,13 +210,56 @@ function* _getFiatAccount({
     providerId,
     fiatConnectProvider.baseUrl
   )
-  const fiatAccount = fiatAccounts.find((account) => account.fiatAccountId === fiatAccountId)
+
+  let fiatAccount: FiatAccount | undefined
+  if (fiatAccountType) {
+    fiatAccount = fiatAccounts.find((account) => account.fiatAccountType === fiatAccountType)
```

Comment:
There's potentially an edge case here, I think.. imagine the scenario where:

* The current quote specifies account type `TypeA` allowing the underlying fiat account schema `SchemaA`
* The user has a `TypeA` account on file with the provider, but it uses `SchemaB`

This method would return the `SchemaB` account since it matches the account type `TypeA`, but it wouldn't actually be valid to use for the given quote since the account schema isn't legal. We should probably also check that the schema matches the legal schemas for the quote as well.

Reply:
good call 👍